### PR TITLE
Fix remaining nonexistent AdamOptions::beta1 calls in C++ frontend tutorial

### LINKED ISSUE #47351: no member named 'beta1' in 'torch::optim::AdamOptions'
## To Reproduce

1. In https://pytorch.org/tutorials/advanced/cpp_frontend.html
1. torch::optim::Adam generator_optimizer(
    generator->parameters(), torch::optim::AdamOptions(2e-4).beta1(0.5));
1.The error: no member named 'beta1' in 'torch::optim::AdamOptions'

## Environment

Please copy and paste the output from our
[environment collection script](https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py)
(or fill out the checklist below manually).

You can get the script and run it with:
```
wget https://raw.githubusercontent.com/pytorch/pytorch/master/torch/utils/collect_env.py
# For security purposes, please check the contents of collect_env.py before running it.
python collect_env.py
```

 - PyTorch Version (e.g., 1.0):
 - OS (e.g., Linux):
 - How you installed PyTorch (`conda`, `pip`, source):
 - Build command you used (if compiling from source):
 - Python version:
 - CUDA/cuDNN version:
 - GPU models and configuration:
 - Any other relevant information:

### PR DESCRIPTION
**Pull request by David-Wu1119**
Fixes pytorch/pytorch#47351

The checkpoint-restore snippet in the C++ frontend tutorial still uses `torch::optim::AdamOptions(2e-4).beta1(0.5)` — the `beta1` member no longer exists (`adam.h` defines only `TORCH_ARG(betas_t, betas)`), so the snippet doesn't compile.

#2260 fixed the identical calls earlier in the tutorial (training-loop section) but missed these two occurrences in the checkpoint-restore block. This applies the same `betas(std::make_tuple(0.5, 0.5))` form to the remaining two, making the file consistent.

*(Prepared with AI assistance; reviewed by the author.)*

### CODE REVIEW

File: advanced_source/cpp_frontend.rst [hunk 1]
```diff
@@ -1207,9 +1207,9 @@ optimizers are created, but before the training loop:
 .. code-block:: cpp
 
   torch::optim::Adam generator_optimizer(
-      generator->parameters(), torch::optim::AdamOptions(2e-4).beta1(0.5));
+      generator->parameters(), torch::optim::AdamOptions(2e-4).betas(std::make_tuple(0.5, 0.5)));
   torch::optim::Adam discriminator_optimizer(
-      discriminator->parameters(), torch::optim::AdamOptions(2e-4).beta1(0.5));
+      discriminator->parameters(), torch::optim::AdamOptions(2e-4).betas(std::make_tuple(0.5, 0.5)));
```
[the remark this answers is not in the document]
Comment (David-Wu1119) [reply advanced_source/cpp_frontend.rst]: Good catch on the learning rate — fixed in `25ed93f`. The checkpoint-restore snippet's discriminator was `2e-4` while the training loop above uses `5e-4`; the two blocks are now byte-identical, which is the point of that snippet (it reconstructs the same optimizers).

On beta2 I went the other way deliberately. `.beta1(0.5)` did leave beta2 at 0.999, but the training-loop block this snippet mirrors already uses `betas(std::make_tuple(0.5, 0.5))` (lines 969/971, set by #2260). Using `(0.5, 0.999)` here would make the restored optimizers differ from the ones actually used for training — a worse inconsistency than the one it fixes. Matching #2260's existing convention keeps the tutorial self-consistent.
